### PR TITLE
Make more classes implement JSONSerializer to reduce boilerplate code

### PR DESCRIPTION
**Pull request by ckelleyRH**
In following packages:

com.netscape.certsrv.system
com.netscape.certsrv.selftests

### CODE REVIEW

File: base/common/src/test/java/com/netscape/certsrv/system/TPSConnectorDataTest.java [hunk 1]
```diff
@@ -33,7 +35,7 @@ public void testJSON() throws Exception {
         String json = before.toJSON();
         System.out.println("JSON (before): " + json);
 
-        TPSConnectorData afterJSON = TPSConnectorData.fromJSON(json);
+        TPSConnectorData afterJSON = JSONSerializer.fromJSON(json, TPSConnectorData.class);
         System.out.println("JSON (after): " + afterJSON.toJSON());
```
Comment (edewata) [inline base/common/src/test/java/com/netscape/certsrv/system/TPSConnectorDataTest.java:39]: Similar issue as in the other PR, we're not supposed to use `JaxbAnnotationIntrospector` anymore, but please verify that the new JSON mapping produces the same result as before.

Comment (ckelleyRH) [reply base/common/src/test/java/com/netscape/certsrv/system/TPSConnectorDataTest.java:39]: Seems OK to me
```
##JacksonAnnotationIntrospector
JSON (before): {
  "id" : "tps0",
  "userID" : "user1"
}
JSON (after): {
  "id" : "tps0",
  "userID" : "user1"
}

##JaxbAnnotationIntrospector
JSON (before): {
  "id":"tps0",
  "UserID":"user1"
}
JSON (after): {
  "id":"tps0",
  "UserID":"user1"
}

Comment (ckelleyRH) [reply base/common/src/test/java/com/netscape/certsrv/system/TPSConnectorDataTest.java:39]: Rebased onto #3598 so it uses `JacksonAnnotationIntrospector` now

**Review by edewata — APPROVED** [review]
LGTM. Thanks!

**Pull request merged**